base/applications/mspaint/main.cpp

```diff
@@ -115,193 +82,171 @@ OFNHookProc(HWND hwnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
+        CImage::GetImporterFilterString(strFilter, aguidFileTypesI, strAllPictureFiles,
+                                        CImage::excludeDefaultLoad, _T('\0'));
+
+        // Initializing the OPENFILENAME structure for GetOpenFileName
+        ZeroMemory(&ofn, sizeof(ofn));
+        ofn.lStructSize    = sizeof(ofn);
+        ofn.hwndOwner      = m_hWnd;
+        ofn.hInstance      = hProgInstance;
+        ofn.lpstrFilter    = strFilter;
+        ofn.Flags          = OFN_EXPLORER | OFN_HIDEREADONLY;
+        ofn.lpstrDefExt    = L"png";
+    }
+
+    ofn.lpstrFile = pszFile;
+    ofn.nMaxFile  = cchMaxFile;
+    return ::GetOpenFileName(&ofn);
+}
+
+BOOL CMainWindow::DoGetSaveFileName(IN OUT LPTSTR pszFile, INT cchMaxFile)
+{
+    static OPENFILENAME sfn = { 0 };
+    static CString strFilter;
 
-int WINAPI
-_tWinMain (HINSTANCE hThisInstance, HINSTANCE hPrevInstance, LPTSTR lpszArgument, INT nCmdShow)
+    if (sfn.lStructSize == 0)
+    {
+        // Get the export filter
+        CSimpleArray<GUID> aguidFileTypesE;
+        CImage::GetExporterFilterString(strFilter, aguidFileTypesE, NULL,
+                                        CImage::excludeDefaultSave, _T('\0'));
+
+        // Initializing the OPENFILENAME structure for GetSaveFileName
+        ZeroMemory(&sfn, sizeof(sfn));
+        sfn.lStructSize    = sizeof(sfn);
+        sfn.hwndOwner      = m_hWnd;
+        sfn.hInstance      = hProgInstance;
+        sfn.lpstrFilter    = strFilter;
+        sfn.Flags          = OFN_EXPLORER | OFN_OVERWRITEPROMPT | OFN_ENABLEHOOK;
+        sfn.lpfnHook       = OFNHookProc;
+        sfn.lpstrDefExt    = L"png";
+
+        // Choose PNG
+        for (INT i = 0; i < aguidFileTypesE.GetSize(); ++i)
+        {
+            if (aguidFileTypesE[i] == Gdiplus::ImageFormatPNG)
+            {
+                sfn.nFilterIndex = i + 1;
+                break;
+            }
+        }
+    }
+
+    sfn.lpstrFile = pszFile;
+    sfn.nMaxFile  = cchMaxFile;
+    return ::GetSaveFileName(&sfn);
+}
+
+BOOL CMainWindow::DoChooseColor(IN OUT COLORREF *prgbColor)
 {
-    HWND hwnd;               /* This is the handle for our window */
-    MSG messages;            /* Here messages to the application are saved */
-
-    HMENU menu;
-    HACCEL haccel;
-
-    TCHAR sfnFilename[1000];
-    TCHAR sfnFiletitle[256];
-    TCHAR ofnFilename[1000];
-    TCHAR ofnFiletitle[256];
-    TCHAR miniaturetitle[100];
-    static COLORREF custColors[16] = {
+    static CHOOSECOLOR choosecolor = { 0 };
+    static COLORREF custColors[16] =
+    {
         0xffffff, 0xffffff, 0xffffff, 0xffffff, 0xffffff, 0xffffff, 0xffffff, 0xffffff,
         0xffffff, 0xffffff, 0xffffff, 0xffffff, 0xffffff, 0xffffff, 0xffffff, 0xffffff
     };
 
+    if (choosecolor.lStructSize == 0)
+    {
+        // Initializing the CHOOSECOLOR structure for ChooseColor
+        ZeroMemory(&choosecolor, sizeof(choosecolor));
+        choosecolor.lStructSize    = sizeof(choosecolor);
```

Comment:
you can remove the extra spaces at the left of the equal sign (for those three lines).

Reply:
Removed.